content/precise-types.md

```diff
@@ -0,0 +1,517 @@
+HkTypeParam  ::=  {Annotation} [‘precise’] [‘+’ | ‘-’] (id [HkTypeParamClause] | ‘_’) TypeBounds ;
+~~~ 
+
+#### Precise Types
+A type is considered to be precise when at least of one of following occurs:
+  * It is a type parameter declaration that is directly modified with `precise` 
+  ~~~ scala
+  //`T` is a precise type parameter declaration
+  class PreciseBox[precise T]
+  ~~~
+  * It is a type variable reference with a precise type parameter origin
+  ~~~ scala
+  //`: T` type variable is precise because it has the precise origin type parameter `T` 
+  def id[precise T](t: T): Box[T] = ??? 
+  val x = id(1) //: Box[1]
+  ~~~
+  * It is substituting a precise type parameter (may also be a wildcard substitution)
+  ~~~ scala
+  def id[precise T](t: T): Box[T] = ???
+  def idBoxBox[BB](x: Box[BB]): Box[BB] = ???
+  //`BB` is precise because it substitutes the precise type parameter `T` 
+  val bb1 = idBoxBox(id(1)) //: Box[1] 
+  ~~~
+  * It is an Alias of a precise type parameter.
+  ~~~ scala
+  class PreciseBox[precise T]
+  //`A` is precise because it is aliasing the precise type parameter `T`
+  type PB[A] = PreciseBox[A]
+  ~~~
+  * It is introduced by an applied type and it is in a precise position of that type. 
+    A precise position can present directly, when the position parameter is annotated with `precise`, or indirectly by "boring" through an applied type composition so that all applied type parameters are precise if the applied type itself is in a precise position.
+  ~~~ scala
+  class PreciseBox[precise T]
+  //`BB` is precise because it was introduced by `PreciseBox` in the precise position `T`
+  def pbox[BB](x: PreciseBox[BB]): Box[BB] = ??? 
+  //`BB` is imprecise because it was not introduced by `PreciseBox` and is not annotated with precise
+  def pbox[BB](b: BB)(x: PreciseBox[BB]): Box[BB] = ??? 
+  class Cov[+T]
+  //`BB` is precise because it was introduced by `Cov` which is applied in `PreciseBox` in the precise position `T`
+  def pboxcov[BB](x: PreciseBox[Cov[BB]]): Box[BB] = ??? 
+  ~~~
+  * It is a upper-bounded by precise type.
+    Upper-bounds may manifest either directly in a type param declaration or dynamically constrained by applying one method as an argument of another.
+  ~~~ scala
+  //`T` is precise because it is upper-bounded by `P`
+  def id[precise P, T <: P](t: T): Box[T] = ??? 
+  ~~~
+  * It is a by-name reference with a precise result type
+  ~~~ scala
+  //`: => T` is precisely typed because the result `T` is precise
+  def id[precise T](t: => T): Box[T] = ??? 
+  ~~~
+  * It is a FuncXX type with a precise result type
+  ~~~ scala
+  //`: (Int, Int) => T` is precisely typed because the result `T` is precise
+  def id[precise T](t: (Int, Int) => T): Box[T] = ??? 
+  ~~~
+
+#### Precise Term Arguments
+A term expression is *precisely typed* when it is applied in a term argument position that is annotated by a precise type. 
```

Comment:
As specified, the term is only precise when applied in a position annotated by a precise type, so the unions/intersections ruling are deferred to that earlier definition (and thus won't be precise, since unions/intersections are not).